packages/dev/core/src/Materials/Node/Blocks/PBR/pbrMetallicRoughnessBlock.ts

```diff
@@ -1068,11 +1074,16 @@ export class PBRMetallicRoughnessBlock extends NodeMaterialBlock {
         const normalShading = this.perturbedNormal;
 
         let worldPosVarName = this.worldPosition.associatedVariableName;
+        let worldNormalVarName = this.worldNormal.associatedVariableName;
         if (this.generateOnlyFragmentCode) {
             worldPosVarName = state._getFreeVariableName("globalWorldPos");
             state._emitFunction("pbr_globalworldpos", `vec3${state.fSuffix} ${worldPosVarName};\n`, comments);
             state.compilationString += `${worldPosVarName} = ${this.worldPosition.associatedVariableName}.xyz;\n`;
 
+            worldNormalVarName = state._getFreeVariableName("globalWorldNormal");
+            state._emitFunction("pbr_globalworldnorm", `vec3${state.fSuffix} ${worldNormalVarName};\n`, comments);

```

Comment:
Do we need this line? I don't see "pbr_globalworldnorm" used anywhere in the code?

Note: it's the same for "pbr_globalworldpos".

Reply:
yeah this is over protective I agree. I'll remove that